breakwater/src/prometheus_exporter.rs

```diff
@@ -45,39 +25,39 @@ impl PrometheusExporter {
     pub fn new(
         listen_addr: &str,
         statistics_information_rx: broadcast::Receiver<StatisticsInformationEvent>,
-    ) -> Result<Self, Error> {
-        let listen_addr = listen_addr.parse().context(ParseListenAddressSnafu {
-            listen_address: listen_addr.to_string(),
-        })?;
-
-        prometheus_exporter::start(listen_addr).context(StartPrometheusServerSnafu)?;
+    ) -> eyre::Result<Self> {
+        let listen_addr = listen_addr.parse()?;
```

Comment:
Or was this change intentional?
IMHO this is one of the benefits on snafu/thiserror, it forces you wrap every error with a meaningful error message :)
```suggestion
        let listen_addr = listen_addr
            .parse()
            .with_context(|| format!("failed to parse listen_addr: {listen_addr}"))?;
```

Reply:
this was intentional. Looking at the message of the `AddrParseError` it felt like there is no value in adding context.

```
% cargo run -- -p '888.777.666.333:1234'                                                                                                            25-03-12 - 11:15:00
    Finished `dev` profile [optimized + debuginfo] target(s) in 0.12s
     Running `target/debug/breakwater -p '888.777.666.333:1234'`
[2025-03-12T10:15:20Z INFO  breakwater::server] Started Pixelflut server on [::]:1234
Error: 
   0: failed to start prometheus exporter
   1: invalid socket address syntax

Location:
   breakwater/src/prometheus_exporter.rs:29

Backtrace omitted. Run with RUST_BACKTRACE=1 environment variable to display it.
Run with RUST_BACKTRACE=full to include source snippets.
```

Reply:
but you are right. It surely feels a bit inconsistent.